content/en/docs/contributing/docs/style-guide.md

```diff
@@ -0,0 +1,400 @@
+
+Term | Usage
+:--- | :----
+Flux | Flux should always be capitalized.
+On-premises | On-premises or On-prem rather than On-premise or other variations.
+Multi-tenancy | Multi-tenancy or Multi-tenant rather than Multitenancy or other variations.
+
+## Shortcodes
+
+Hugo [Shortcodes](https://gohugo.io/content-management/shortcodes) help create different rhetorical appeal levels.
+
+1. Surround the text with an opening and closing shortcode.
+
+2. Use the following syntax to apply a style:
+
+   ```go-html-template
+   {{</* note */>}}
+   No need to include a prefix; the shortcode automatically provides one. (Note:, Caution:, etc.)
+   {{</* /note */>}}
+   ```
+
+   The output is:
+
+   {{< note >}}
+   The prefix you choose is the same text for the tag.
+   {{< /note >}}
+
+### Note
+
+Use `{{</* note */>}}` to highlight a tip or a piece of information that may be helpful to know.
+
+For example:
+
+```go-html-template
+{{</* note */>}}
+You can _still_ use Markdown inside these callouts.
+{{</* /note */>}}
+```
+
+The output is:
+
+{{< note >}}
+You can _still_ use Markdown inside these callouts.
+{{< /note >}}
+
+You can use a `{{</* note */>}}` in a list:
+
+```go-html-template
+1. Use the note shortcode in a list
+
+1. A second item with an embedded note
+
+   {{</* note */>}}
+   Warning, Caution, and Note shortcodes, embedded in lists, need to be indented four spaces. See [Common Shortcode Issues](#common-shortcode-issues).
+   {{</* /note */>}}
+
+1. A third item in a list
+
+1. A fourth item in a list
+```
+
+The output is:
+
+1. Use the note shortcode in a list
+
+1. A second item with an embedded note
+
+    {{< note >}}
+    Warning, Caution, and Note shortcodes, embedded in lists, need to be indented four spaces. See [Common Shortcode Issues](#common-shortcode-issues).
+    {{< /note >}}
+
+1. A third item in a list
+
+1. A fourth item in a list
+
+### Caution
+
+Use `{{</* caution */>}}` to call attention to an important piece of information to avoid pitfalls.
+
+For example:
+
+```go-html-template
+{{</* caution */>}}
+The callout style only applies to the line directly above the tag.
+{{</* /caution */>}}
+```
+
+The output is:
+
+{{< caution >}}
+The callout style only applies to the line directly above the tag.
+{{< /caution >}}
+
+### Warning
+
+Use `{{</* warning */>}}` to indicate danger or a piece of information that is crucial to follow.
+
+For example:
+
+```go-html-template
+{{</* warning */>}}
+Beware.
+{{</* /warning */>}}
+```
+
+The output is:
+
+{{< warning >}}
+Beware.
+{{< /warning >}}
+
+## Markdown elements
+
+### Line breaks
+
+Use a single newline to separate block-level content like headings, lists, images, code blocks, and others. The exception is second-level headings, where it should be two newlines. Second-level headings follow the first-level (or the title) without any preceding paragraphs or texts. A two line spacing helps visualize the overall structure of content in a code editor better.
```

Comment:
Does this mean use a single newline at the end of a paragraph, or put a single, extra newline between paragraphs? (This document does the former where there are consecutive paras, mostly, but the advice is ambiguous, and I much prefer the latter!)

Reply:
I'm actually quite a fan of https://sembr.org/ in markdown. It makes git diffs much nicer when the primary content is human language. Any thoughts on this? In all projects where I've seen this adopted it's helped a lot.

Reply:
After having a look at this, I want to remove the second level heading part.

I would like to look at paragraph formatting a bit further into the future.

There is already a bunch of new guidelines in this PR already, and I think it would be for the best to see how things go with this implemented for a few weeks, then add on new improvements.

Let me know if I'm overthinking this a bit too much.